### PR TITLE
feat(outputs): add outputs proxy to the toolkit

### PR DESCRIPTION
**Pull request by abouroubi**
**Why?**

Solves

#119 

**How?**

Used a Proxy like the inputs object

---

- [x] Tests have been added/updated (if necessary)
- [x] Documentation has been updated (if necessary)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -30,7 +31,7 @@ export interface ToolkitOptions {
   token?: string
 }
 
-export class Toolkit<I extends InputType = InputType> {
+export class Toolkit<I extends InputType = InputType, O extends OutputType = OutputType> {
```
Comment (JasonEtco) [inline src/index.ts:34]: Love this! ✨

File: src/outputs.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as core from '@actions/core'
+
+export interface OutputType { [key: string]: string | undefined }
+
+export function createOutputProxy <O extends OutputType = OutputType>() {
+  return new Proxy<O>({} as O, {
+    set (_, name: string, value: string): boolean {
+      // When we attempt to get `outputs.___`, instead
```
Comment (JasonEtco) [inline src/outputs.ts:8]:
```suggestion
      // When we attempt to set `outputs.___`, instead
```

**Comment by abouroubi** [comment]
Yes the state of the output parameters isn't exposed by the actions core SDK, we can add an object to keep it in the outputs proxy.
Even if I'm not sure if it will be really usefull, I can't see why we need to check if we already have a value for an output, and it's probably why the core SDK don't expose it

**Comment by abouroubi** [comment]
Thanks @JasonEtco .

Do you have an approximation date on when it will be released ?

**Pull request merged**